crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1497,50 +1497,60 @@ where
 /// Once the [RpcModule] is built via [RpcModuleBuilder] the servers can be started, See also
 /// [ServerBuilder::build] and [Server::start](jsonrpsee::server::Server::start).
 #[derive(Default, Debug)]
-pub struct RpcServerConfig {
+pub struct RpcServerConfig<Http = Identity, Rpc = Identity, Endpoint = String> 
```

Comment:
@mattsse i think this is closer to what you had in mind, is that right?

Reply:
yeah but we need a simpler way to configure this for server configs at the same time, so we don't support configuring different layers for different server types, because this would be too complex.

we can either store the layers as separate fields and keep identity for all servers or add helper functions that apply the layer for all the servers.